libindy/src/commands/non_secrets.rs

```diff
@@ -129,7 +129,14 @@ impl NonSecretsCommandExecutor {
 
         self._check_type(type_)?;
 
-        let res = self.wallet_service.add_record(wallet_handle, type_, id, value, tags_json.unwrap_or("{}"))?; //TODO: question
+        let tags = match tags_json {
+            None => HashMap::new(),
+            Some(tags_string) =>  match serde_json::from_str(tags_string) {
+                Ok(tag_names) => tag_names,
+                Err(serde_json_err) => return Err(IndyError::WalletError(WalletError::InputError(format!("Invalid tags input: {}", tags_string))))
+            }
+        };
```

Comment:
we always return CommonInvalidStructure in case of failed parsing (on commands layer) of input json parameter.